### PR TITLE
DATA RESULT

### PR DESCRIPTION
**Pull request by tchasinga**
```
ALTER TABLE animals ADD COLUMN species VARCHAR(255);
```
Screenshot (260)

```
Insert the provided data:

INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Charmander', '2020-02-08', 0, false, -11.0, 'Fire');

INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Plantmon', '2021-11-15', 2, true, -5.7, 'Grass');

INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Squirtle', '1993-04-02', 3, false, -12.13, 'Water');

INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Angemon', '2005-06-12', 1, true, -45.0, 'Angel');

INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Boarmon', '2005-06-07', 7, true, 20.4, 'Mammal');

INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Blossom', '1998-10-13', 3, true, 17.0, 'Flower');

INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
VALUES ('Ditto', '2022-05-14', 4, true, 22.0, 'Shape-shifter');

```
Screenshot (262)

```
-- Begin the transaction for changing species to unspecified
BEGIN;

-- Update species for all animals to 'Unspecified'
UPDATE animals SET species = 'Unspecified';

-- Verify changes before rollback
SELECT * FROM animals;

-- Rollback the transaction
ROLLBACK;

-- Verify species after rollback
SELECT * FROM animals;

-- Begin the main transaction
BEGIN;

-- Update species for animals whose name ends in "mon"
UPDATE animals SET species = 'Digimon' WHERE name LIKE '%mon';

-- Update species for animals with no species set
UPDATE animals SET species = 'Pokemon' WHERE species IS NULL;

-- Verify changes before committing
SELECT * FROM animals;

-- Commit the main transaction
COMMIT;

-- Verify changes after commit
SELECT * FROM animals;


```
Screenshot (269)
Screenshot (270)
Screenshot (271)
Screenshot (268)

```
-- Begin the transaction
BEGIN;

-- Delete all records from the animals table
DELETE FROM animals;

-- Roll back the transaction
ROLLBACK;

-- Verify records after rollback
SELECT * FROM animals;
```

Screenshot (264)

```
-- Begin the transaction
BEGIN;

-- Delete animals born after Jan 1st, 2022
DELETE FROM animals WHERE date_of_birth > '2022-01-01';

-- Create a savepoint
SAVEPOINT my_savepoint;

-- Update weights to be negative
UPDATE animals SET weight_kg = weight_kg * -1;

-- Rollback to the savepoint
ROLLBACK TO my_savepoint;

-- Update negative weights
UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;

-- Commit the transaction
COMMIT;
```

Screenshot (265)

```

1. **How many animals are there?**

SELECT COUNT(*) AS total_animals FROM animals;

2. **How many animals have never tried to escape?**

SELECT COUNT(*) AS no_escape_animals FROM animals WHERE escape_attempts = 0;

3. **What is the average weight of animals?**

SELECT AVG(weight_kg) AS average_weight FROM animals;

4. **Who escapes the most, neutered or not neutered animals?**

SELECT neutered, MAX(escape_attempts) AS max_escape_attempts
FROM animals
GROUP BY neutered;

5. **What is the minimum and maximum weight of each type of animal?**

SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
FROM animals
GROUP BY species;

6. **What is the average number of escape attempts per animal type of those born between 1990 and 2000?**

SELECT species, AVG(escape_attempts) AS avg_escape_attempts
FROM animals
WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
GROUP BY species;
```

Screenshot (275)
Screenshot (276)

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -23,3 +23,95 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+UPDATE animals SET species = 'Digimon' WHERE name LIKE '%mon';
+
+-- Update species for animals with no species set
+UPDATE animals SET species = 'Pokemon' WHERE species IS NULL;
+
+-- Verify changes before committing
+SELECT * FROM animals;
+
+-- Commit the transaction
+COMMIT;
```
Comment (AbelG101) [inline queries.sql:42]: - [ ] It seems like you're missing the transaction for the question "Inside a transaction update the animals table by setting the species column to unspecified. Verify that the change was made. Then roll back the change and verify that the species columns went back to the state before the transaction." which should be before the highlighted SQL transaction.
- [ ] You're screenshot for this transaction is also incomplete so please update it.

File: queries.sql [hunk 2]
```diff
@@ -23,3 +23,95 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY species;
+
+6. **What is the average number of escape attempts per animal type of those born between 1990 and 2000?**
+
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
```
Comment (AbelG101) [inline queries.sql:117]: - [ ] Kindly add screenshots for the above queries to ensure compliance with the project requirements.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @tchasinga 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Species column is properly added ✔
- Proper transaction management ✔
- Correct use of Git Flow ✔
- Descriptive PR title with a great screenshot summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by christianbiring1 — APPROVED** [review]
Hi @tchasinga , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 

## Highlights
 - [ ] Great work on adding the screen shot as required by the previous reviewer :+1: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**